### PR TITLE
[Datx SWR] Readme updates

### PR DESCRIPTION
**Pull request by isBatak**

### CODE REVIEW

File: docs/jsonapi-swr/api.md [hunk 1]
```diff
@@ -107,16 +119,16 @@ const getKey = (pageIndex: number, previousPageData: CollectionResponse) => {
   return getPageExpression(pageIndex);
 };
 
-const = useDatx(getKey, config);
+const = useDatxInfinite(getKey, config);
```
Comment (kristian240) [inline docs/jsonapi-swr/api.md:122]: Missing variable

```suggestion
const datx = useDatxInfinite(getKey, config);
```

File: docs/jsonapi-swr/overview.md [hunk 2]
```diff
@@ -217,6 +209,66 @@ const { data: user } = useDatx(getUserQuery(id));
...
+
+// single item lazy query, a.k.a currying
+const getTodoQuery = (id?: string) => () =>
+  id
+    ? ({
+        id,
+        op: 'getOne',
+        type: 'todos',
+      } as const satisfies IGetOneExpression<typeof Todo>)
+    : null;
```
Comment (kristian240) [inline docs/jsonapi-swr/overview.md:247]: As discussed, I would remove this in favor of using the `useSWR` API. `useDatx(() => getTodoQuery(todo.id))`

File: docs/jsonapi-swr/api.md [hunk 3]
```diff
@@ -47,8 +47,7 @@ Second parameter of `useDatx` is for passing config options. It extends default
 
 ### Expression signature
 
-Currently we support 3 expressions for fetching resources `getOne`, `getMany` and `getAll`.
-Future plan is to support generic `request` operation and `getRelated`.
+Currently, we support 4 expressions for fetching resources `getOne`, `getMany`, `getAll` and `getRelatedResource`.
```
Comment (doroz0) [inline docs/jsonapi-swr/api.md:50]: There's one more - `getRelatedResources`

File: docs/jsonapi-swr/api.md [hunk 4]
```diff
@@ -73,10 +72,23 @@ export interface IGetAllExpression<TModel extends JsonapiModelType = JsonapiMode
...
+// fetch related resource through the primary resource
+export interface IGetRelatedResourceExpression<
+  TModelType extends JsonapiModelType = JsonapiModelType,
+> {
+  readonly op: 'getRelatedResource';
+  readonly type: TModelType['type'];
+  readonly relation: string;
+  id: string;
+  queryParams?: IRequestOptions['queryParams'];
+}
```
Comment (doroz0) [inline docs/jsonapi-swr/api.md:85]: One more

```
export interface IGetRelatedResourcesExpression<
    TModelType extends JsonapiModelType = JsonapiModelType
> {
    readonly op: 'getRelatedResources';
    readonly type: TModelType['type'];
    readonly relation: string;
    id: string;
    queryParams?: IRequestOptions['queryParams'];
}
```

File: docs/jsonapi-swr/overview.md [hunk 5]
```diff
@@ -176,45 +170,109 @@ export const todosQuery = {
...
+
+// related query through primary resource
+const getPostCommentsRelationshipQuery = (postId?: string) =>
+  postId
+    ? ({
+        id: postId,
+        op: 'getRelationship',
+        relationship: 'comments',
+        type: 'posts',
+      } as const satisfies IGetRelationshipExpression<typeof Post>)
```
Comment (doroz0) [inline docs/jsonapi-swr/overview.md:207]: Did you mean `getRelatedResources` or is this something else?

```
const getPostCommentsRelationshipQuery = (postId?: string) =>
  postId
    ? ({
        id: postId,
        op: 'getRelatedResources',
        relation: 'comments',
        type: 'posts',
      } as const satisfies IGetRelatedResourcesExpression<typeof Post>)
    : null;
```

File: docs/jsonapi-swr/overview.md [hunk 6]
```diff
@@ -176,45 +170,109 @@ export const todosQuery = {
...
+    : null;
+
+const getPostCommentsRelationshipQuery = (postId?: string) =>
+  postId
+    ? ({
+        id: postId,
+        op: 'getRelationship',
+        relationship: 'comments',
+        type: 'posts',
+      } as const satisfies IGetRelationshipExpression<typeof Post>)
```
Comment (doroz0) [inline docs/jsonapi-swr/overview.md:247]: Here also

**Pull request merged**